src/app/portlets/dot-edit-page/content/components/dot-edit-page-workflows-actions/dot-edit-page-workflows-actions.component.ts

```diff
@@ -13,20 +14,47 @@ export class DotEditPageWorkflowsActionsComponent implements OnInit {
     @Input() inode: string;
     @Input() label: string;
 
-    workflowsActions: Observable<MenuItem[]>;
+    workflowsMenuActions: Observable<MenuItem[]>;
 
-    constructor(private workflowsService: DotWorkflowService) {}
+    constructor(private workflowsService: DotWorkflowService, private httpErrorManagerService: DotHttpErrorManagerService) {}
 
     ngOnInit() {
-        this.workflowsActions = this.workflowsService
-            .getContentWorkflowActions(this.inode)
-            .map((workflows: DotWorkflowAction[]) => this.getWorkflowOptions(workflows));
+        this.workflowsMenuActions = this.workflowsService.getContentWorkflowActions(this.inode).map((workflows: DotWorkflowAction[]) => {
+            return this.getWorkflowOptions(workflows);
+        });
     }
 
     private getWorkflowOptions(workflows: DotWorkflowAction[]): MenuItem[] {
         return workflows.map((workflow: DotWorkflowAction) => {
             return {
-                label: workflow.name
+                label: workflow.name,
+                command: () => {
+                    const currentMenuActions = this.workflowsMenuActions;
+                    this.workflowsMenuActions = this.workflowsService
+                        .fireWorkflowAction(this.inode, workflow.id)
+                        .pluck('inode')
+                        .catch((error) => Observable.of(null))
+                        .mergeMap((inode: string) => {
+                            this.inode = inode || this.inode;
+                            return this.workflowsService.getContentWorkflowActions(this.inode).map((newWorkflows: DotWorkflowAction[]) => {
+                                if (newWorkflows.length === 0) {
```

Comment:
Please add a TODO here  beacause this will probably change if the endpoint gets updated.

Reply:
when the end point return 'newWorkflows.length === 0'? when the inode doesn't exists?

Reply:
Because the endpoint returns a 200 with an empty array, I already told the Jons that in that case should return a 404

Reply:
but the empty array means that the inode doesn't have any actions or that the inode doesn't exists

Reply:
Yes, but if we send an inode that doesn't exist still return 200 with []